### PR TITLE
[Housekeeping] Review all the pending TODOs comments related with .NET 8

### LINKED ISSUE #8989: Reconcile all NET8 //TODO Add this API comments
### Description

Because we need to update the published api txt files and we don't want to add any new/breaking changes with servicing. We've just been adding //TODO NET8 notes for any APIs that we plan on making public.

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Review all the pending TODOs comments related with .NET 8.

Fixes #8989

### CODE REVIEW

**Comment by mattleibow** [comment]
These all need a discussion on each type. Not sure we can do breaking changes at this time.

File: src/Core/src/Handlers/Editor/EditorHandler.Android.cs [hunk 1]
```diff
@@ -7,14 +7,11 @@
 
 namespace Microsoft.Maui.Handlers
 {
-	// TODO: NET8 issoto - Change the TPlatformView generic type to MauiAppCompatEditText
-	// This type adds support to the SelectionChanged event
-	public partial class EditorHandler : ViewHandler<IEditor, AppCompatEditText>
+	public partial class EditorHandler : ViewHandler<IEditor, MauiAppCompatEditText>
```
Comment (rmarinho) [inline src/Core/src/Handlers/Editor/EditorHandler.Android.cs:10]: I would not make this change, and just remove the comments.

File: src/Controls/src/Core/Platform/iOS/Extensions/FormattedStringExtensions.cs [hunk 2]
```diff
@@ -53,7 +53,7 @@ public static class FormattedStringExtensions
 		public static NSAttributedString ToNSAttributedString(
 			this Span span,
 			IFontManager fontManager,
-			double defaultLineHeight = 0d, // TODO: NET8 should be -1, but too late to change for NET8
+			double defaultLineHeight = -1,
```
Comment (rmarinho) [inline src/Controls/src/Core/Platform/iOS/Extensions/FormattedStringExtensions.cs:56]: too late for this? maybe move the comment to .NET9 ?

File: src/Core/src/Handlers/Entry/EntryHandler.Android.cs [hunk 3]
```diff
@@ -4,20 +4,19 @@
...
+using Microsoft.Maui.Platform;
 using static Android.Views.View;
 using static Android.Widget.TextView;
 
 namespace Microsoft.Maui.Handlers
 {
-	// TODO: NET8 issoto - Change the TPlatformView generic type to MauiAppCompatEditText
-	// This type adds support to the SelectionChanged event
-	public partial class EntryHandler : ViewHandler<IEntry, AppCompatEditText>
+	public partial class EntryHandler : ViewHandler<IEntry, MauiAppCompatEditText>
```
Comment (rmarinho) [inline src/Core/src/Handlers/Entry/EntryHandler.Android.cs:13]: I would keep like it was and just remove the comments.

File: src/Core/src/Platform/Windows/CalendarDatePickerExtensions.cs [hunk 4]
```diff
@@ -3,8 +3,7 @@
 
 namespace Microsoft.Maui.Platform
 {
-	//TODO make this public on NET8
-	internal static class CalendarDatePickerExtensions
+	public static class CalendarDatePickerExtensions
```
Comment (rmarinho) [inline src/Core/src/Platform/Windows/CalendarDatePickerExtensions.cs:6]: I think we want to keep this internal and just remove the comment.

File: src/Core/src/Platform/Windows/TextBoxExtensions.cs [hunk 5]
```diff
@@ -216,8 +216,7 @@ public static void UpdateSelectionLength(this TextBox textBox, ITextInput entry)
 				textBox.SelectionLength = entry.SelectionLength;
 		}
 
-		// TODO: NET8 issoto - Revisit this, marking this method as `internal` to avoid breaking public API changes
-		internal static int GetCursorPosition(this TextBox textBox, int cursorOffset = 0)
+		public static int GetCursorPosition(this TextBox textBox, int cursorOffset = 0)
```
Comment (rmarinho) [inline src/Core/src/Platform/Windows/TextBoxExtensions.cs:219]: I would keep this one  internal and remove comment

File: src/Core/src/Platform/iOS/TextInputExtensions.cs [hunk 6]
```diff
@@ -3,10 +3,9 @@
 
 namespace Microsoft.Maui.Platform
 {
-	// TODO: NET8 issoto - Revisit this, marking this class as `internal` to avoid breaking public API changes
-	internal static class TextInputExtensions
+	public static class TextInputExtensions
```
Comment (rmarinho) [inline src/Core/src/Platform/iOS/TextInputExtensions.cs:6]: I would keep these internal

**Comment by samhouts** [comment]
Probably should review this one, too: https://github.com/dotnet/maui/issues/8989

File: src/Core/src/Platform/iOS/TextInputExtensions.cs (hunk 6 above)
Comment (jsuarezruiz) [reply src/Core/src/Platform/iOS/TextInputExtensions.cs]: Changed.

**Comment by jsuarezruiz** [comment]
> These all need a discussion on each type. Not sure we can do breaking changes at this time.

Yeah, I have been applying some suggestions but I think that in some cases we need discussion to decide.

**Comment by jsuarezruiz** [comment]
@PureWeen Thoughts?, should we close (for now) this one?

**Comment by samhouts** [comment]
Yeah, I think we're unfortunately too late in the cycle to do these now. These will have to go in NET 9, if at all. Thanks, Javier!

**Pull request closed without merging**